nautobot/docs/development/release-checklist.md

```diff
@@ -1,46 +1,66 @@
 # Release Checklist
 
+This document is intended for Nautobot maintainers and covers the steps to perform when releasing new versions.
+
 ## Minor Version Bumps
 
 ### Update Requirements
 
-Required Python packages are maintained in two files. `base_requirements.txt` contains a list of all the packages required by Nautobot. Some of them may be pinned to a specific version of the package due to a known issue. For example:
+Required Python packages are maintained in two files: `pyproject.toml` and `poetry.lock`. 
+
+#### The `pyproject.toml` file
+Python packages are defined inside of `pyproject.toml`. The `[tool.poetry.dependencies]` section of this file contains a list of all the packages required by Nautobot.
+
+Where possible, we use [tilde requirements](https://python-poetry.org/docs/dependency-specification/#tilde-requirements) to specify a minimal version with some ability to update, for example:
 
 ```
-# https://github.com/encode/django-rest-framework/issues/6053
-djangorestframework==3.8.1
+# REST API framework
+djangorestframework = "~3.12.2"
 ```
 
-The other file is `requirements.txt`, which lists each of the required packages pinned to its current stable version. When Nautobot is installed, the Python environment is configured to match this file. This helps ensure that a new release of a dependency doesn't break Nautobot.
+This would allow Poetry to install `djangorestframework` versions `>=3.12.2` but `<3.13.0`.
+
+#### The `poetry.lock` file
+
+The other file is `poetry.lock`, which is managed by Poetry and contains package names, versions, and other metadata.
+
+Each of the required packages pinned to its current stable version. When Nautobot is installed, this file is used to resolve and install all dependencies listed in `pyproject.toml`, but Poetry will use the exact versions found in `poetry.lock` to ensure that a new release of a dependency doesn't break Nautobot.
+
+!!! warning
+    You must never directly edit this file. You will use `poetry update` commands to manage it.
+
+#### Run `poetry update`
+
+Every minor version release should refresh `poetry.lock`, so that it lists the most recent stable release of each package. To do this:
 
-Every minor version release should refresh `requirements.txt` so that it lists the most recent stable release of each package. To do this:
+1. Run `poetry install` to install the latest version of all required packages.
+2. Run all tests and check that the UI and API function as expected.
+3. Review each requirement's release notes for any breaking or otherwise noteworthy changes.
+4. If a required package requires updating to a new release not covered in the version constraints for a package as defined in `pyproject.toml`, (e.g. `Django ~3.1.7` would never install `Django >=4.0.0`), update it manually in `pyproject.toml`.
+5. Run `poetry update <package>` to update the package versions in `poetry.lock` as appropriate.
```

Comment:
`poetry install` will install the version pinned by `poetry.lock`, not the latest available version. Should this instead be something like:

```suggestion
1. Review each requirement's release notes for any breaking or otherwise noteworthy changes.
2. Run `poetry update <package>` to update the package versions in `poetry.lock` as appropriate.
3. If a required package requires updating to a new release not covered in the version constraints for a package as defined in `pyproject.toml`, (e.g. `Django ~3.1.7` would never install `Django >=4.0.0`), update it manually in `pyproject.toml`.
4. Run `poetry install` to install the refreshed versions of all required packages.
5. Run all tests and check that the UI and API function as expected.
```